ledger/src/blockstore_processor.rs

```diff
@@ -853,12 +853,8 @@ pub fn process_blockstore_from_root(
     // We might be promptly restarted after bad capitalization was detected while creating newer snapshot.
     // In that case, we're most likely restored from the last good snapshot and replayed up to this root.
     // So again check here for the bad capitalization to avoid to continue until the next snapshot creation.
-    if !bank_forks
-        .read()
-        .unwrap()
-        .root_bank()
-        .calculate_and_verify_capitalization(debug_verify)
-    {
+    let bank = bank_forks.read().unwrap().root_bank();
+    if start_slot != bank.slot() && !bank.calculate_and_verify_capitalization(debug_verify) {
```

Comment:
seems unnecessary to change. I suggest just add the `start_slot != bank.slot()` before the existing code '~bank_forks...`

Reply:
and squash the whole thing into that 1 commit ;-) This has turned out to be very simple, thankfully.

Reply:
oh... and you need bank for slot. nevermind.